android/KMEA/app/src/main/java/com/tavultesoft/kmea/data/LanguageResource.java

```diff
@@ -65,10 +67,21 @@ public void setLanguage(String languageID, String languageName) {
 
   /**
    * Helper method if the language resource has an updated kmp package available to download from the cloud
+   * The version in the kmp link is compared to the currentVersion
```

Comment:
Is the version guaranteed to be embedded in the url? I am slightly concerned that we are depending on a side-effect.

Reply:
The url comes from api.keyman.com, and as long as it's a valid package, should always contain a version.

If version is missing, `getQueryParameter("version")` is null and `FileUtils.compareVersions()` returns `FileUtils.VERSION_INVALID`.

Reply:
> The url comes from api.keyman.com, and as long as it's a valid package, should always contain a version.

The URL should be treated as opaque. Just because the version number is in there at present doesn't mean we can rely on that. One day in the future the URL returned from api.keyman.com may return a URL with a hash instead of a version embedded, for example. I just want to avoid adding dependencies on this kind of thing because it does tend to come back to bite us.

For the purposes of clarity, which api are you calling on api.keyman.com to get this URL? I am not entirely clear.

Reply:
> For the purposes of clarity, which api are you calling on api.keyman.com to get this URL? I am not entirely clear.

Currently, the "kmp" link is initially seeded by the query for the lexical-model catalog
https://api.keyman-staging.com/model?q

using the "packageFilename" field.
e.g. `"packageFilename": "https://keyman-staging.com/go/package/download/model/nrc.str.sencoten?version=1.0.5&update=1",`

This catalog is used to generate a list of available lexical models.

Later when the package update checks are run, the "kmp" link gets set to the one from
https://api.keyman-staging.com/package-version?platform=android&model=nrc.str.sencoten

I'm able to do a version check for package-version before determining whether to keep the kmp link

Reply:
So, that api query https://api.keyman-staging.com/model?q returns the model version as `"version": "1.1",` for example, and that's the value you should be storing alongside the model file.

Reply:
Deferring this to #3777